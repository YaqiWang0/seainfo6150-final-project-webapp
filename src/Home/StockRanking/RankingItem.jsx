import React, { useEffect, useState} from "react"
import { useHistory, useLocation } from "react-router-dom";
import style from "./StockRanking.module.css"

const RankingItem = (props) => {
    // TODO: semetic html
    let history = useHistory();
    const openDetail = async (e) => {
        const response = await fetch(`https://financialmodelingprep.com/api/v3/profile/${props.stock.ticker}?apikey=56f622a93b55019c5a0875058fc58f13`);
        const responseJson = await response.json();
        const newsResponse = await fetch(`https://financialmodelingprep.com/api/v3/stock_news?tickers=${props.stock.ticker}&limit=10&apikey=56f622a93b55019c5a0875058fc58f13`);
        const newsResponseJson = await newsResponse.json();
        history.push(
            "/company",
            {
                info: Object.values(responseJson),
                news: newsResponseJson,
                login: props.login
            }
        )
    }

    return (
    <div className={style.ranking_item} onClick={openDetail}>
        <div className={style.name}>
            <span className={style.stock_name}>{props.stock.ticker}</span>
            <span className={style.company_name}>{props.stock.companyName}</span>
        </div>
        <span className={props.isGainer ? style.gainer_price_text : style.loser_price_text}>{`$${props.stock.price}`}</span>
    </div>
    );
}

export default RankingItem