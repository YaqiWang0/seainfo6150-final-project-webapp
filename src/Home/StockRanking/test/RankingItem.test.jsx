import React from "react";
import data from "./data"
import RankingItem from "../RankingItem";

describe("RankingItem test", () => {
    const login = {
        login: true,
        userId: "12345",
        email: "test@test.com"

    }
    it('when the item is not clicked', function () {
        const {container} = render(<RankingItem isGainer={false} stock={data} key={data.ticker} login={login}/>);
        expect(container).toMatchSnapshot();
    })
})