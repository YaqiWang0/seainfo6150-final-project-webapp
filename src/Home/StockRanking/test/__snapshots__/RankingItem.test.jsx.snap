// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingItem test when the item is not clicked 1`] = `
<div>
  <div
    class="ranking_item"
    role="click"
  >
    <div
      class="name"
    >
      <span
        class="stock_name"
      >
        SLDB
      </span>
      <span
        class="company_name"
      >
        Solid Biosciences Inc
      </span>
    </div>
    <span
      class="loser_price_text"
    >
      $5.7706
    </span>
  </div>
</div>
`;
